### ISSUE TITLE
subsetMaf question

### ISSUE THREAD

**Issue by CynStr** [PoisonAlien/maftools#82]
I've downloaded the maf files using the TCGAbiolinks so I could get the data aligned to the hg38 build. These have a longer TCGA 'Tumor_Sample_Barcode' for example TCGA-61-2096-01A-01W-0722-08 as opposed to TCGA-61-2096. 
I could only get the subsetMaf to work with the full tsb, is there a way I could subset the maf file based on the first 12 characters (e.g. TCGA-61-2096) of the tsb?

**Comment by PoisonAlien** [comment]
Hi,

While reading maf with `read.maf` use the argument `isTCGA = TRUE` and it will restrict barcode to first 12 characters.

**Comment by CynStr** [comment]
Great, thanks!

**Comment by ccchang0111** [comment]
Another question regarding subsetMaf's "restrictTo", in code **line 102-105**

```{r}
if(restrictTo != "all"){
    maf.dat = rbind(maf.dat, maf.dat.rest, fill = TRUE, use.names = TRUE)
    maf.silent = rbind(maf.silent, maf.silent.rest, fill = TRUE, use.names = TRUE)
  }
```

Should it be **if(restrictTo == "all)** ? Because if we only restrictTo "cnv" or "mutations", we should not rbind maf.dat and maf.dat.rest. 

Thank you so much for creating all these great modules!

**Comment by PoisonAlien** [comment]
Hi Andrew, 

Thanks for the PR, I will see how.

Regarding the query, when  `restrictTo == 'cnv'` it means all subset operations are applied to only CNV data whereas mutation data remains unchanged and they are stored in `maf.dat.rest` unchanged. Once subsetting is done they are merged back (rbind) together. 
If  `restrictTo == 'all'` which is by default, there is no need to separate the dataset since subset operations will be performed on everything. 

Can you post your commands and where exactly its going wrong ?

**Comment by ccchang0111** [comment]
I see. I probably misunderstood the meaning of `restrictTo == 'cnv'`. Because for one of my work, I combine maf file and cnv file (which only has Hugo_Symbols, SampleID, CNV_call), and sometime I just want to subset cnv samples, so I thought maybe I can use `restrictTo == "cnv"`, and plot subset data with only cnv result. But during coOncoplot, I got everything including other mutation types, which makes the plot a bit confusing.  That is why I thought if I do not `rbind` `maf.dat` and `maf.dat.rest` together, I can get only the cnv subset and plot them.

I guess if want to subset only cnv data, it can be achieved by using `query = "Variant_Type == 'CNV'"`. Maybe this is a better way?

Another issue in the code:

If I only subset cnv file, now the maf.dat only contains cnv samples without other mutation types  (`query = "Variant_Type == 'CNV'"`), then I ran into the following error:

> Error in `[.data.table`(vc.cast, , 2:ncol(vc.cast), with = FALSE) : Item 1 of j is 2 which is outside the column number range [1,ncol=1]

I think the error came from `summarizeMaf`. Because now the dat.maf only has four columns with values (Hugo_Symbol, Variant_Classification, Variant_Type, Tumor_Sample_Barcode), other columns are NA, which cause some problems in `summarizeMaf`.  I have sent another pull-request for summarizeMaf as well. 

Thank you again!

Andrew

**Comment by grantn5** [comment]
Hi @PoisonAlien this is still an issue if you subset a maftools object to only CNVs 

```
maftools::subsetMaf(my_maf, query = "Variant_Type == 'CNV'")

Error in `[.data.table`(vc.cast, , 2:ncol(vc.cast), with = FALSE) : 
  Item 1 of j is 2 which is outside the column number range [1,ncol=1]

```
Is there a solution to this problem?
I am using maftools v2.16.0

**Comment by PoisonAlien** [comment]
Hi Grant,

On my datasets, the function works properly

```r
> maftools::subsetMaf(maf, query = "Variant_Type == 'CNV'")
-Processing clinical data
An object of class  MAF 
           ID summary      Mean Median
       <char>   <num>     <num>  <num>
1: NCBI_Build      NA        NA     NA
2:     Center      NA        NA     NA
3:    Samples     179        NA     NA
4:     nGenes      39        NA     NA
5:      total       0 0.0000000      0
6:        Amp     255 1.4245810      1
7:        Del      99 0.5530726      0
8:  CNV_total     354 1.9776536      1
```

This issue has been addressed in #908 
Could you please update the package from GitHub and try again?
